### ISSUE TITLE
1.6.2 Release

### ISSUE THREAD

**Issue by vjeux** [prettier/prettier#2804]
I'm pretty busy with work but we should release 1.6.2 with what we have:
- JSX tweaks
- pragma
- TypeScript edge cases
- CSS cases

Is anyone interested in writing the changelog and doing the release?

**Comment by kentor** [comment]
this is going in? https://github.com/prettier/prettier/pull/2722

**Comment by vjeux** [comment]
@kentor yeah, everything being merged in is going in.

**Comment by suchipi** [comment]
I can do it this evening

**Comment by vjeux** [comment]
@suchipi awesome!

**Comment by suchipi** [comment]
I'll need to be a collaborator on npm in order to do the actual release. So I can either just do the changelog, or you if you want to make me a collaborator, I can do the release, too.

**Comment by existentialism** [comment]
I can help with the changelog too, @suchipi if you need a hand

**Comment by vjeux** [comment]
You're been added :) https://www.npmjs.com/package/prettier/access

**Comment by suchipi** [comment]
Cool :)

**Comment by suchipi** [comment]
Draft here: https://github.com/prettier/prettier/releases/edit/untagged-3768fef8696e3930b05c

Any feedback before release?

Also, when running build, I get:
```
...
How to test against dist:
  1) yarn test --prod

How to publish:
  1) IMPORTANT!!! Go to dist/
  2) npm publish
✨  Done in 61.70s.
```
But `yarn test --prod` doesn't work:
```
$ yarn test --prod
yarn test v1.0.1
$ jest "--prod"
/Users/suchipi/Code/prettier/node_modules/jest-util/build/validateCLIOptions.js:62
    throw createCLIValidationError(unrecognizedOptions, allowedOptions);
    ^

● Unrecognized CLI Parameter:

  Unrecognized option "prod".

  CLI Options Documentation:
  http://facebook.github.io/jest/docs/cli.html

error Command failed with exit code 1.
```

Is there a new way to run the tests against `dist/`? (They pass when run normally).

**Comment by vjeux** [comment]
Just to make sure, did you run `yarn` to update all the dependencies?

**Comment by azz** [comment]
I think yarn v1 behaves differently, try `yarn --prod test`

**Comment by existentialism** [comment]
Release notes LGTM, I can just update https://github.com/prettier/prettier/pull/2811 to link to it instead.

**Comment by azz** [comment]
Maybe drop the big prettier logo for a "patch" release?

**Comment by suchipi** [comment]
I was thinking... since we're *technically* adding features, should it be 1.7.0?

**Comment by suchipi** [comment]
Also sorry about the duplicated work @existentialism 🙏

**Comment by existentialism** [comment]
@suchipi Oh, no worries at all! I figured we were doing simpler release notes for this one like the previous patch releases.

**Comment by vjeux** [comment]
Yeah, there's a bunch of things in the release so would make more sense to be 1.7. Could you add three more highlights:

- JSX changes (the main reason why we're doing an early release)
- TypeScript bug fix that's affecting people
- We now lowercase all the things in CSS

**Comment by suchipi** [comment]
> Just to make sure, did you run `yarn` to update all the dependencies? - @vjeux 

Yeah, I blew away my package.json and reran `yarn install` to be sure

>I think yarn v1 behaves differently, try `yarn --prod test`
>
>With yarn v1 you don't need `--` to separate arguments, so the `--prod` is being passed to jest instead. - @azz 
`yarn --prod test` appears to do a `yarn install`... 😅 

I can just downgrade my local yarn for now

**Comment by vjeux** [comment]
Would be nice to do the actual release tomorrow morning, so we can be available if anything goes wrong. I don't want to stress out about things that could break during the night :p

**Comment by suchipi** [comment]
Updated draft: https://github.com/prettier/prettier/releases/edit/untagged-268c192abfe88cff3597

@vjeux I wasn't sure which typescript issue you were referring to, but I assumed it was the `never` type one

**Comment by vjeux** [comment]
Awesome, looks good to me :)

**Comment by existentialism** [comment]
FWIW, I know the declare enum's been asked for a bit as well.

**Comment by suchipi** [comment]
I'll give it a shout-out

**Comment by suchipi** [comment]
Updated: https://github.com/prettier/prettier/releases/edit/untagged-35331d37ecb417dd86a7

**Comment by mitermayer** [comment]
@suchipi can you also highlight `--config-precedence` since this is an important update for editor integrations

**Comment by suchipi** [comment]
Done :) https://github.com/prettier/prettier/releases/edit/untagged-8efb2e4a79466b492146

**Comment by lydell** [comment]
Regarding CSS letter case: In `display: BLOCK;`, "BLOCK" will remain unchanged. `DISPLAY: block;` will turn into `display: block;`, though.

**Comment by azz** [comment]
I've just moved a couple of bits around to look more like previous release notes. https://github.com/prettier/prettier/releases/edit/untagged-0a9ddfefa7ed61f6c674

**Comment by vjeux** [comment]
Whenever you are ready :)

**Comment by lydell** [comment]
Now that we went from 1.6.2 (patch release) to 1.7.0 (minor release), I think it is an as good as time as any to merge #2767 and release it in 1.7.0. Thoughts?

I've corrected the CSS case example, btw: https://github.com/prettier/prettier/releases/edit/untagged-ec958688c7b4247a1771

**Comment by suchipi** [comment]
Published! :) https://github.com/prettier/prettier/releases/tag/1.7.0

**Comment by vjeux** [comment]
@suchipi can you tweet about it so that I can retweet?

**Comment by suchipi** [comment]
will do

**Comment by suchipi** [comment]
https://twitter.com/suchipi/status/908381181623771136

**Comment by suchipi** [comment]
Thanks for the help in putting the release together, everybody 😊

**Comment by suchipi** [comment]
Looks like playground is broken? https://prettier.io/playground/
"TypeError: process.version is undefined"

**Comment by vjeux** [comment]
Seems to be working for me on Chrome mac, but it still says 1.6.1

**Comment by vjeux** [comment]
Same issue here. cc @lydell

**Comment by lydell** [comment]
@suchipi @vjeux Seems to be a problem with the new version when built for the web, not the new Playground features. I can look into it anyway, though.

**Comment by suchipi** [comment]
Which part of the process deploys the website, by the way? I ran `yarn build`, made and pushed a 1.7.0 commit on master (which had website changes from running yarn build), then went into dist and ran npm publish. I didn't see any pre or post publish hooks, so I'm guessing we have a github hook setup that publishes github pages when master changes?

**Comment by vjeux** [comment]
`yarn build` updates the `website/` folder.

**Comment by lydell** [comment]
@prettier-bot updates the website.

**Comment by lydell** [comment]
It was the update of cosmiconfig that broke stuff, btw. It uses [please-upgrade-node](https://github.com/typicode/please-upgrade-node) which looks at `process.version`, which we hadn't shimmed before.

**Comment by suchipi** [comment]
Thanks for figuring it out

**Comment by suchipi** [comment]
Looks good now; thanks @lydell!
